src/curl.cpp

```diff
@@ -2944,7 +2975,7 @@ void S3fsCurl::insertV4Headers(const std::string& access_key_id, const std::stri
     get_date_sigv3(strdate, date8601);
 
     std::string contentSHA256 = payload_hash.empty() ? EMPTY_PAYLOAD_HASH : payload_hash;
-    const std::string realpath = pathrequeststyle ? "/" + S3fsCred::GetBucket() + server_path : server_path;
+    const std::string realpath = pathrequeststyle ? extractURI(s3host) + S3fsCred::GetBucket() + server_path : server_path;
```

Comment:
As clang-tidy points out, make this method static.

Reply:
@ggtakec the method is now static 👍